scripts/ci/setup-dev-sso.sh

```diff
@@ -0,0 +1,289 @@
+     loginWithEmailAllowed: true
+     duplicateEmailsAllowed: false
+     resetPasswordAllowed: false
+     editUsernameAllowed: false
+     bruteForceProtected: false
+     permanentLockout: false
+     maxFailureWaitSeconds: 900
+     minimumQuickLoginWaitSeconds: 60
+     waitIncrementSeconds: 60
+     quickLoginCheckMilliSeconds: 1000
+     maxDeltaTimeSeconds: 43200
+     failureFactor: 30
+     clients:
+       - id: 9a5018a7-5f92-40c9-b8f1-63f53bc32a68
+         clientId: sandbox-public
+         surrogateAuthRequired: false
+         enabled: true
+         clientAuthenticatorType: client-secret
+         redirectUris:
+           - '*'
+         webOrigins:
+           - '*'
+         notBefore: 0
+         bearerOnly: false
+         consentRequired: false
+         standardFlowEnabled: true
+         implicitFlowEnabled: false
+         directAccessGrantsEnabled: true
+         serviceAccountsEnabled: false
+         publicClient: true
+         frontchannelLogout: false
+         protocol: openid-connect
+         protocolMappers: []
+         attributes: {}
+         authenticationFlowBindingOverrides: {}
+         fullScopeAllowed: true
+         nodeReRegistrationTimeout: -1
+         defaultClientScopes: []
+         optionalClientScopes: []
+     clientScopes: []
+     defaultDefaultClientScopes: []
+     smtpServer: {}
+     loginTheme: rh-sso
+     eventsEnabled: false
+     eventsListeners:
+       - jboss-logging
+     enabledEventTypes: []
+     adminEventsEnabled: false
+     adminEventsDetailsEnabled: false
+     identityProviders: []
+     identityProviderMappers: []
+     internationalizationEnabled: false
+     supportedLocales: []
+     authenticationFlows: []
+     authenticatorConfig: []
+     userManagedAccessAllowed: false
+     users:
+       - credentials:
+           - type: password
+             value: user1
+         email: user1@user.us
+         emailVerified: true
+         enabled: true
+         firstName: user1
+         id: user1
+         username: user1
+         clientRoles: {}"
+    echo "objects to be created in order to install operator"
+    cat <<EOF | oc apply -f -
+${INSTALL_KEYCLOAK}
+EOF
+
+while ! oc get statefulset -n ${DEV_SSO_NS} keycloak &> /dev/null ; do
+    printf "waiting for keycloak statefulset in ${DEV_SSO_NS} to be ready...\n"
+    sleep 10
+done
+
+printf "waiting for keycloak in ${DEV_SSO_NS} to be ready...\n"
+ATTEMPT=0
+MAX_NUM_ATTEMPTS=100
+SLEEP_TIME=1
+while [[ -z $(oc get keycloak sandbox-dev -o jsonpath='{.status.ready}' -n ${DEV_SSO_NS} | grep "true" || true) ]]; do
+    echo "$(( ATTEMPT++ )). attempt (out of ${MAX_NUM_ATTEMPTS}) of waiting for keycloak sandbox-dev/${DEV_SSO_NS} to be ready"
+    sleep ${SLEEP_TIME}
+    if [[ ${ATTEMPT} -eq ${MAX_NUM_ATTEMPTS} ]]; then
+      echo "reached timeout of waiting for keycloak to be available in the cluster - see following info for debugging:"
+      oc get keycloak sandbox-dev -n ${DEV_SSO_NS} -o yaml
+    fi
+done
```

Comment:
we may want to simplify using `oc wait`. Something similar to this may work

```
TIMEOUT=100s
oc wait --for jsonpath='{.status.ready}'=true keycloak/sandbox-dev -n "${DEV_SSO_NS}" --timeout "${TIMEOUT}" \
    || { oc get keycloak sandbox-dev -n ${DEV_SSO_NS} -o yaml && exit 1 }
```

Reply:
good point, I've applied your suggestion in https://github.com/codeready-toolchain/toolchain-cicd/pull/113/commits/07761694b9f83d5e089c2a366f10a487e7934216 . Thanks!